src/parser/grammar.ne

```diff
@@ -23,11 +23,20 @@ const toKeywordNode = (token: Token): KeywordNode => ({
   raw: token.raw,
 });
 
-const addLeadingComments = (node: AstNode, comments: CommentNode[]): AstNode =>
-  comments.length > 0 ? { ...node, leadingComments: comments } : node;
-
-const addTrailingComments = (node: AstNode, comments: CommentNode[]): AstNode =>
-  comments.length > 0 ? { ...node, trailingComments: comments } : node;
+interface CommentAttachments {
+  leading?: CommentNode[];
+  trailing?: CommentNode[];
+}
+
+const addComments = (node: AstNode, { leading, trailing = [] }: CommentAttachments): AstNode => {
+  if (leading && leading.length > 0) {
```

Comment:
can be
```
return {
  ...node,
  leading: leading?.length ? leading : undefined,
  trailing: trailing?.length ? trailing : undefined,
}
```

Reply:
Thought of doing this originally, but wanted to completely avoid empty fields on the nodes.

But using `leading?.length` syntax would definitely improve things.

Reply:
Added PR with fixes #465